arviz/data/inference_data.py

```diff
@@ -177,6 +177,11 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def __iter__(self):
+        for group in self._groups_all:  # pylint: disable=protected-access
+            dataset = getattr(self, group)
```

Comment:
I think we can assume all groups are found as an attribute, so nothing needs to be changed here.

If user add stuff `_groups_all` let it fail.